fancontrol.py

```diff
@@ -31,25 +27,24 @@ def get_temp():
         raise RuntimeError('Could not parse temperature output.')
 
 
-if __name__ == '__main__':
-    # Validate the on and off thresholds
-    if OFF_THRESHOLD >= ON_THRESHOLD:
-        raise RuntimeError('OFF_THRESHOLD must be less than ON_THRESHOLD')
 
-    fan = OutputDevice(GPIO_PIN)
+def get_speed():
+    """
+    Dynamically change the output of the motor
+    Returns: A percentage of max max speed corresponding to temperature
+    """
+    speed = 0.0
+    for i in range(len(TEMPS)):  # loop through list of temps
+        if get_temp() > TEMPS[i]:
+            speed = OUTPUTS[i]  # set speed to the corresponding temp
+    return speed
 
-    while True:
-        temp = get_temp()
 
-        # Start the fan if the temperature has reached the limit and the fan
-        # isn't already running.
-        # NOTE: `fan.value` returns 1 for "on" and 0 for "off"
-        if temp > ON_THRESHOLD and not fan.value:
-            fan.on()
+if __name__ == '__main__':
 
-        # Stop the fan if the fan is running and the temperature has dropped
-        # to 10 degrees below the limit.
-        elif fan.value and temp < OFF_THRESHOLD:
-            fan.off()
+    fan = PWMOutputDevice(GPIO_PIN)
+
+    while True:
+        fan.value = get_speed()
 
-        time.sleep(SLEEP_INTERVAL)
+        time.sleep(SLEEP_INTERVAL)
```

Comment:
I see here I didn't leave a line at the end of the code and its raised an error. What is the reason for needing a new line at the end of the code?

Reply:
This is a standard for most programming languages. It helps ensure proper parsing on all platforms. Here's one explanation: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file